File: expenses-web/src/services/accountService.ts
import {apiRequest} from "@/lib/apiClient.ts";
import type {AccountDto, CreateAccountDto} from "@/types/accountDto";

const API_URL = '/api/account';

const getAllAccounts = (): Promise<AccountDto[]> => {
    return apiRequest<AccountDto[]>(API_URL, 'GET');
};

const getAccountById = (id: number): Promise<AccountDto> => {
    return apiRequest<AccountDto>(`${API_URL}/${id}`, 'GET');
};

const createAccount = (accountData: CreateAccountDto): Promise<AccountDto> => {
    return apiRequest<AccountDto>(API_URL, 'POST', accountData);
};

const deactivateAccount = (id: number): Promise<AccountDto> => {
    return apiRequest<AccountDto>(`${API_URL}/${id}/deactivate`, 'PATCH');
}

const updateAccount = (id: number, accountData: Partial<AccountDto>): Promise<AccountDto> => {
    return apiRequest<AccountDto>(
        `${API_URL}/${id}/edit`,
        'PATCH',
        accountData
    );
}

const updateBalance = (id: number, newBalance: number): Promise<AccountDto> => {
    return apiRequest<AccountDto>(
        `${API_URL}/${id}/balance`,
        'PATCH',
        newBalance
    );
}

const healthCheck = (): Promise<string> => {
    return apiRequest<string>(`${API_URL}/health`, 'GET');
}

export const accountService = {
    getAllAccounts,
    getAccountById,
    createAccount,
    deactivateAccount,
    updateAccount,
    updateBalance,
    healthCheck
}

File: expenses-back/src/main/kotlin/br/com/money/expenses/service/AccountService.kt
package br.com.money.expenses.service

import br.com.money.expenses.model.dto.account.AccountDto

interface AccountService {
   fun getAllAccounts(): List<AccountDto>

   fun getAccountById(id: Long): AccountDto

   fun createAccount(
      name: String,
      balance: Double,
      color: String
   ): AccountDto

   fun deactivateAccount(id: Long) : AccountDto

   fun updateAccount(
      id: Long,
      name: String,
      color: String
   ) : AccountDto

   fun updateBaseAmount(
      id: Long,
      newBaseAmount: Double
   ) : AccountDto
}